### ISSUE TITLE
Strange issue with Automapper 4 and overloaded constructors

### ISSUE THREAD

**Issue by dplaskon** [AutoMapper/AutoMapper#815]
Since moving to Automapper 4, I've seen some strange behavior in mapped objects that have overloaded constructors.

Here is a (somewhat) simple repro to show the issue:

```vb
        <Test()>
        Public Sub Can_Reproduce_Automapper_Constructor_Issue()

            Mapper.CreateMap(Of Tester, Tester2)()

            Dim t As New Tester()

            Dim dest = Mapper.Map(Of Tester2)(t)

        End Sub

        <Serializable()>
        Public Class Tester

            Public Sub New()

            End Sub

            Public Property UserId As Integer


        End Class

        <Serializable()>
        Public Class Tester2

            Public Sub New()

            End Sub

            Public Sub New(ByVal userId As Integer,
                           ByVal active As Boolean)
                Me.UserId = userId
                Me.Active = active
            End Sub

            Public Property UserId As Integer

            Public Property Active As Boolean

        End Class
```

The following exception is thrown:

> AutoMapper.AutoMapperMappingException : Missing type map configuration or unsupported mapping.

> Mapping types:
> Tester -> Boolean
> MyProject.IntegrationTests.TasksTests.TempTests+Tester -> System.Boolean

> Destination path:
> Tester2

> Source value:
> MyProject.IntegrationTests.TasksTests.TempTests+Tester

Removing the overloaded constructor results in successful mapping.

Any ideas?

**Comment by TylerCarlson1** [comment]
Submitted a PR to fix this in #816.

Issue is now have can resolve for constructor parameters, but they aren't accounted for correctly.

**Comment by dplaskon** [comment]
Thanks :)
